src/NuGet.Core/NuGet.ProjectModel/PackageSpecWriter.cs

```diff
@@ -526,23 +532,39 @@ private static void SetFrameworkReferences(IObjectWriter writer, ISet<FrameworkD
             }
         }
 
-        private static void SetCentralDependencies(IObjectWriter writer, ICollection<CentralPackageVersion> centralPackageVersions)
+        private static void SetCentralDependencies(IObjectWriter writer, ICollection<CentralPackageVersion> centralPackageVersions, bool compressed)
         {
             if (!centralPackageVersions.Any())
             {
                 return;
             }
 
-            writer.WriteObjectStart("centralPackageVersions");
+            var cpvmElementName = "centralPackageVersions";
+            if (compressed)
```

Comment:
How about we change what we write in the assets file/dg spec? 

That's an internal representation and the reader/writer can be built in a backwards compatible way.

Reply:
@nkolev92 would you mind adding more details to help me understand your thinking/concerns?

I was tinkering with the idea to persist to disk only a hash for Central Package versions (or maybe not persist even that information)  "in the majority" of cases. 
I used "majority" because for nuget.exe to work the temp dgspec needs to contain all the CPV information. So writers and readers need to have the ability to write and read CPV information.
Also this will mean to break the principle that the serialized dgspec can be used as input for a restore operation. 

For the above two reasons I would like to try achieving better performance with less radical changes before moving towards the above idea.

Reply:
The only *relevant* information that a PackageVersion carries is id and version right? 
As the type suggests: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.LibraryModel/CentralPackageVersion.cs. 

Writing a json such as:
```json
{
  "a" : "1.0.0",
  "b" : "2.0.0"
}

should net us some wins.

Reply:
This is is exactly what is written. One sample:
``` 
centralPackageVersions": {
          "LibTestCM1": "[3.0.0, )",
          "Newtonsoft.Json": "[12.0.3, )",
          "ParentLibTestCM1": "[3.0.0, )"
        }```

Reply:
I think I'd suggest to do this regardless, eliminating the `compressed` concept. It can be made backwards compatible on the reader side (see how dependencies are read). 

Thoughts?

Reply:
Discussed offline. 
1. The versions needs to be serialized as they are now in order to support nuget.exe scenarios. 
2. The GetHash(bool compressed) was removed and the GetHash() will create a compressed hash.